### DISCUSSION TITLE
Center windows when focusing windows that don't fit

### DISCUSSION THREAD

**Discussion by tversteeg** [YaLTeR/niri#95] [Ideas]
I have an idea about an option that automatically centers the focused window when another window doesn't fit next to it.

So for example, I have the following three windows, where the percentages are the size of the window as a percentage of the screen size:

<table>
<tr>
 <td>A
 <td>B
 <td>C
<tr>
 <td>80%
 <td>50%
 <td>50%
</table>

Now when I'm focused on *A*, and I call `focus-column-right`, I would like the *B* window to be focused and centered, because it doesn't fit together with *A* on a single screen anymore, 80% + 50% > 100%.

When I'm focused on *B* and I call `focus-column-right` again, I would like the *B* window to be on the left side of the screen and the *C* window on the right, because they both fit, 50% + 50% <= 100%.

**Comment by YaLTeR** [comment]
Hm, interesting. I know PaperWM has an option to always center, but not something like this I think.

**Reply by tversteeg** [reply]
I've created a PR for this in #96, this adds both an option to always center and also what I described above.
